src/Polly.Core/Simmy/Outcomes/OutcomeChaosPipelineBuilderExtensions.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace Polly.Simmy.Outcomes;
+
+/// <summary>
+/// Extension methods for adding outcome to a <see cref="ResiliencePipelineBuilder"/>.
+/// </summary>
+public static partial class OutcomeChaosPipelineBuilderExtensions
+{
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="fault">The exception to inject.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder AddChaosFault(this ResiliencePipelineBuilder builder, bool enabled, double injectionRate, Exception fault)
+    {
+        Guard.NotNull(builder);
+
+        builder.AddFaultCore(new OutcomeStrategyOptions<Exception>
+        {
+            Enabled = enabled,
+            InjectionRate = injectionRate,
+            Outcome = new(fault)
+        });
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="faultGenerator">The exception generator delegate.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder AddChaosFault(
+        this ResiliencePipelineBuilder builder, bool enabled, double injectionRate, Func<ValueTask<Outcome<Exception>?>> faultGenerator)
+    {
+        Guard.NotNull(builder);
+
+        builder.AddFaultCore(new OutcomeStrategyOptions<Exception>
+        {
+            Enabled = enabled,
+            InjectionRate = injectionRate,
+            OutcomeGenerator = (_) => faultGenerator()
+        });
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="options">The fault strategy options.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder AddChaosFault(this ResiliencePipelineBuilder builder, OutcomeStrategyOptions<Exception> options)
```

Comment:
```suggestion
    public static ResiliencePipelineBuilder AddChaosResult(this ResiliencePipelineBuilder builder, OutcomeStrategyOptions options)
```

You can still inject exception as outcome represents both result and exception, or not?

``` csharp
new ResiliencePipelineBuilder().
   .AddChaosResult(new()
  {
     OutcomeGenerator => Outcome.FromExceptionAsTask<object>(new InvalidOperationException())
  })
  .Build();
```

Or what do you think about having pure non-reactive strategy dedicated to injecting exceptions? 

I know we talked about avoiding duplications but something about `OutcomeStrategyOptions<Exception>` doesn't feel right. It tells me (purely based on the other APIs) that this is reactive strategy for callbacks that returns exception in.

Reply:
> You can still inject exception as outcome represents both result and exception, or not?

You could still do so from the `AddChaosResult` extension

> Or what do you think about having pure non-reactive strategy dedicated to injecting exceptions?
I know we talked about avoiding duplications but something about OutcomeStrategyOptions<Exception> doesn't feel right. It tells me (purely based on the other APIs) that this is reactive strategy for callbacks that returns exceptions.

I get your point but honestly, IMO I don't think that's necessary since Simmy users are already familiar with that, plus the context is clear because of the extension name (addChaosFault), not to mention that when you look at the types it's also clear what it does:

* Non-reactive strategy that injects a fault
![image](https://github.com/App-vNext/Polly/assets/12032625/852cf295-22a1-4672-9777-0d7913d1b38e)
* Reactive strategy that injects a faukt
![image](https://github.com/App-vNext/Polly/assets/12032625/5df0bf2d-2078-4ddf-9266-04760d938c46)

Reply:
> It would save you from duplicating AddChaosFault to both builders. Just target ResiliencePipelineBuilderBase. The API could be cleaner too. The downside would be functionality overlap, but we could keep the outcome injection only to generic builder.

Not following this, can you provide an example please?

Reply:
> Not following this, can you provide an example please?

Given these options for fault-injection strategy:

``` csharp
public class FaultStrategyOptions: MonkeyStrategyOptions
{
    public Func<OnFaultInjectedArguments, ValueTask>? OnFaultInjected { get; set; }

    public Func<FaultGeneratorArguments, ValueTask<Exception?>>? FaultGenerator { get; set; }

    public Exception? Fault { get; set; }
}
```

You could have the following extension that targets both builders:

``` csharp
    // convenience extensions here

    public static TBuilder AddFault<TBuilder>(this TBuilder builder, FaultStrategyOptions options)
        where TBuilder : ResiliencePipelineBuilderBase
    {
        ...
    }
```

Your outcome strategy could be a little cleaner too since now it does not deal with faults, only outcomes.

Reply:
So you suggest creating a `FaultStrategyOptions` to separate it from `OutcomeStrategyOptions`?

Reply:
It's just an idea, since injecting exceptions is non-reactive behavior in general.

Reply:
Great idea!